tcms/telemetry/api.py

```diff
@@ -171,9 +171,30 @@ def _append_status_counts_to_result(count, result):
         result.get(status).append(status_count)
 
 
-@rpc_method(name='Testing.test_case_health')
-def test_case_health(query=None):
+@rpc_method(name='Testing.test_case_health.grouped')
+def test_case_health_grouped(query=None):
+    data = test_case_health(query)
+
+    result = {
+        'flaky': [],
+        'all_fail': [],
+        'almost_stable': []
+    }
 
+    for element in data:
+        failed_percent = element['count']['fail'] / element['count']['all']
+        if failed_percent == 1:
+            result['all_fail'].append(element)
+        elif failed_percent >= 0.5:
+            result['flaky'].append(element)
+        elif failed_percent < 0.5:
+            result['almost_stable'].append(element)
+
```

Comment:
Sort elements in the lists before returning.

Reply:
Note: I will re-think the band distribution which will necessitate update to the numeric constants used in this function and also in `_build_result_from_data()`.

I'm wondering if there's a possibility to have these ranges defined somewhere and use them. How about tuples (lower, upper) ?

Reply:
here, the elements are already sorted. e.g. `data` is sorted, because it comes from `test_case_health` and `test_case_health` is sorting before returning:
```python
def test_case_health(query=None):
    # more code

    data.sort(key=_sort_by_passing_rate)

    return _build_result_from_data(data)
```
or do you mean sort by something else?

Reply:
> I'm wondering if there's a possibility to have these ranges defined somewhere and use them. 

Sure, we can define them whenever we decide. config files, env variables, you name it. 

> How about tuples (lower, upper) ?

I don't understand that part of the comment